pkg/codegen/nodejs/gen.go

```diff
@@ -2525,3 +2527,39 @@ export function resourceOptsDefaults(): any {
 	_, err := fmt.Fprintf(w, body, pluginDownloadURL)
 	contract.AssertNoError(err)
 }
+
+func genInstallScript(pluginDownloadURL string) string {
+	const installScript = `"use strict";
+var childProcess = require("child_process");
+
+var args = process.argv.slice(2);
+
+// NOTE: version split is intended to prevent sed style replacement
+if (args.indexOf("${VERS" + "ION}") != -1) {
```

Comment:
Clever. Can we get away without it, though, and just look for the `"${VERSION}"` string as-is? I don't think we're doing sed style replacement in this file anywhere. And if there is a `pluginDownloadURL` set, the URL could contain `${VERSION}` with #8507, so it's possible this file may end up having `${VERSION}` in it anyway.

Reply:
I think that https://github.com/pulumi/pulumi/pull/8507 ensures that you are right here. Good catch.